modules.d/01systemd-cryptsetup/module-setup.sh

```diff
@@ -0,0 +1,198 @@
+
+    [[ $hostonly ]] || [[ $mount_needs ]] && {
+        for fs in "${host_fs_types[@]}"; do
+            [[ $fs = "crypto_LUKS" ]] && return 0
+        done
+        return 255
+     }
+
+    # If the module dependency requirements are not fulfilled
+    # return 1 to not include the required module(s).
+
+    if ! dracut_module_included "dm"; then
+        derror "systemd-cryptsetup needs dm in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "rootfs-block"; then
+        derror "systemd-cryptsetup needs rootfs-block in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "rootfs-block"; then
+        derror "systemd-cryptsetup needs rootfs-block in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "systemd"; then
+        derror "systemd-cryptsetup needs systemd in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "systemd-ask-password"; then
+        derror "systemd-cryptsetup needs systemd-ask-password in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "systemd-tmpfiles"; then
+        derror "systemd-cryptsetup needs tmpfiles in the initramfs."
+        return 1
+    fi
+
+    # Return 255 to only include the module, if another module requires it.
+    return 255
+
+}
+
+# Module dependency requirements.
+depends() {
+
+    # This module has external dependency on the systemd module.
+    echo dm rootfs-block systemd systemd-ask-password systemd-tmpfiles
+    # Return 0 to include the dependent systemd module in the initramfs.
+    return 0
+
+}
+
+# Install kernel required file(s) for the module in the initramfs.
+installkernel() {
+        hostonly="" instmods drbg
+        instmods dm_crypt
+
+        # in case some of the crypto modules moved from compiled in
+        # to module based, try to install those modules
+        # best guess
+        [[ $hostonly ]] || [[ $mount_needs ]] && {
+            # dmsetup returns s.th. like
+            # cryptvol: 0 2064384 crypt aes-xts-plain64 :64:logon:cryptsetup:....
+            dmsetup table | while read name _ _ is_crypt cipher _; do
+            [[ $is_crypt != "crypt" ]] && continue
+                # get the device name
+                name=/dev/$(dmsetup info -c --noheadings -o blkdevname ${name%:})
+                # check if the device exists as a key in our host_fs_types
+                if [[ ${host_fs_types[$name]+_} ]]; then
+                    # split the cipher aes-xts-plain64 in pieces
+                    _OLD_IFS=$IFS
+                    IFS='-:'
+                    set -- $cipher
+                    IFS=$_OLD_IFS
+                    # try to load the cipher part with "crypto-" prepended
+                    # in non-hostonly mode
+                    hostonly= instmods $(for k in "$@"; do echo "crypto-$k";done)
+                fi
+        done
+        }
+    return 0
+}
+
+# called by dracut
+cmdline() {
```

Comment:
depend on `crypt` and remove this

Reply:
Module adjusted to depend on crypt and it removed